plugin.js

```diff
@@ -1,445 +1,402 @@
-(function (tinymce) {
-  'use strict'
-
-  tinymce.PluginManager.requireLangPack('map')
-
-  var SERVICES = ['google_staticmap', 'google_embed']
-  SERVICES['google_staticmap'] = {
-    name: 'Google Static Map',
-    api: '//maps.googleapis.com/maps/api/staticmap?',
-    type: 'img'
-  }
-  SERVICES['google_embed'] = {
-    name: 'Google Embed Map',
-    api: '//www.google.com/maps/embed/v1/place?',
-    type: 'iframe'
-  }
-  var MARKER_REGEX = /^(?:color:)?([^|]*)(?:\|.*)?$/
-
-  function buildURL (url, params) {
-    var key
-    var value
-    var pairs = []
+(function mapPlugin(tinymce) {
+  "use strict";
+
+  tinymce.PluginManager.requireLangPack("map");
+
+  const SERVICES = ["google_staticmap", "google_embed"];
+  SERVICES["google_staticmap"] = {
+    name: "Google Static Map",
+    api: "//maps.googleapis.com/maps/api/staticmap?",
+    type: "img"
+  };
+  SERVICES["google_embed"] = {
+    name: "Google Embed Map",
+    api: "//www.google.com/maps/embed/v1/place?",
+    type: "iframe"
+  };
+  const MARKER_REGEX = /^(?:color:)?([^|]*)(?:\|.*)?$/;
+
+  function buildURL(url, params) {
+    const pairs = [];
+    let key, value;
 
     for (key in params) {
-      value = params[key]
+      value = params[key];
       if (value) {
-        pairs.push(key + '=' + value)
+        pairs.push(key + "=" + value);
       }
     }
-    url += pairs.join('&')
-    return url
+    url += pairs.join("&");
+    return url;
   }
 
   // parseUri 1.2.2
   // (c) Steven Levithan <stevenlevithan.com>
   // MIT License
-  function parseUri (str) {
-    var o = parseUri.options
-    var m = o.parser[o.strictMode ? 'strict' : 'loose'].exec(str)
-    var uri = {}
-    var i = 14
-
-    while (i--) uri[o.key[i]] = m[i] || ''
+  function parseUri(str) {
```

Comment:
Browsers now have the `URL` construct that can be used to parse. And there is `URLSearchParams` that can give us the query string. Maybe we switch to that instead?